gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/SpecStoreChangeMonitor.java

```diff
@@ -74,7 +76,15 @@ public String load(String key) throws Exception {
   protected GobblinServiceJobScheduler scheduler;
 
   public SpecStoreChangeMonitor(String topic, Config config, int numThreads) {
-    super(topic, config, numThreads);
+    // Differentiate group id for each host
+    super(topic, config.withValue(GROUP_ID_KEY,
+        ConfigValueFactory.fromAnyRef(SPEC_STORE_CHANGE_MONITOR_PREFIX + UUID.randomUUID().toString())),
+        numThreads);
+  }
+
+  @Override
+  protected void initializeConsumerClient() {
```

Comment:
Add comment here to mention that we expect the client to consume all partitions in the topic from the latest offset? You can even move that part of logic here

Reply:
I can add that comment to explain it but it is difficult to move that logic here as I need access to the underlying kafkaConsumer itself to get the Partition Info for a particular kafka topic and then convert it to the list of kafka partitions I need to set it to. That logic is more easily managed in the consumer client itself.